apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.Data/SpannerCommand.cs

```diff
@@ -0,0 +1,367 @@
+        private int _commandTimeout;
+        private SpannerTransaction _transaction;
+
+        /// <summary>
+        /// </summary>
+        public SpannerCommand()
+        {
+            DesignTimeVisible = true;
+            _commandTimeout = (int)ConnectionPoolOptions.Instance.Timeout.TotalSeconds;
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="commandTextBuilder"></param>
+        /// <param name="connection"></param>
+        /// <param name="transaction"></param>
+        /// <param name="parameters"></param>
+        public SpannerCommand(SpannerCommandTextBuilder commandTextBuilder, SpannerConnection connection,
+            SpannerTransaction transaction = null, SpannerParameterCollection parameters = null)
+        {
+            GaxPreconditions.CheckNotNull(commandTextBuilder, nameof(commandTextBuilder));
+            GaxPreconditions.CheckNotNull(connection, nameof(connection));
+
+            SpannerCommandTextBuilder = commandTextBuilder;
+            SpannerConnection = connection;
+            _transaction = transaction;
+            Parameters = parameters;
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="commandText"></param>
+        /// <param name="connection"></param>
+        /// <param name="transaction"></param>
+        /// <param name="parameters"></param>
+        public SpannerCommand(string commandText, SpannerConnection connection, SpannerTransaction transaction = null,
+            SpannerParameterCollection parameters = null)
+            : this(SpannerCommandTextBuilder.FromCommandText(commandText), connection, transaction, parameters)
+        {
+        }
+
+        /// <inheritdoc />
+        public override string CommandText
+        {
+            get { return SpannerCommandTextBuilder?.ToString() ?? string.Empty; }
+            set { SpannerCommandTextBuilder = SpannerCommandTextBuilder.FromCommandText(value); }
+        }
+
+        /// <inheritdoc />
+        public override int CommandTimeout
+        {
+            get { return _commandTimeout; }
+            set { _commandTimeout = value; }
+        }
+
+        /// <inheritdoc />
+        public override CommandType CommandType
+        {
+            get { return CommandType.Text; }
+            set
+            {
+                if (!Equals(value, CommandType.Text))
+                    throw new NotSupportedException("Cloud Spanner only supports CommandType.Text.");
+            }
+        }
+
+        /// <inheritdoc />
+        public override bool DesignTimeVisible { get; set; }
+
+        /// <summary>
+        /// </summary>
+        public new SpannerParameterCollection Parameters { get; private set; }
+
+        /// <summary>
+        /// </summary>
+        public SpannerConnection SpannerConnection { get; set; }
+
+        /// <inheritdoc />
+        public override UpdateRowSource UpdatedRowSource
```

Comment:
No.  Cloud Spanner does not support multiple SQL commands within a single command.  This has an effect on autoincrement columns -- but which are frowned upon with UUIDs being recommended practice.  For now, I disallow anything but None with a descriptive error, but needs some followup on how to update a dataset after an insert with autoincrement.  For now, it might have to be an additional manual query.